lib/foreman_tasks/dynflow/console_authorizer.rb

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+module ForemanTasks
+  class Dynflow::ConsoleAuthorizer
+    def initialize(env)
+      @rack_request          = Rack::Request.new(env)
+      @user_id, @expires_at = @rack_request.session.values_at('user', 'expires_at')
+      @user                 = User.find_by_id(@user_id) unless session_expired?
+    end
+
+    def allow?
+      if @user && (unlimited_edit? || authorized_for_task?)
+        return true
+      else
+        return false
```

Comment:
Just to explain how that come to be there (in case you wondered), this is how the method looked before, so I missed the last refactoriing step. Thanks for notifying this. Fixed now

```ruby
def authorized_console?(env)
  rack_request        = Rack::Request.new(env)
  user_id, expires_at = rack_request.session.values_at('user', 'expires_at')

  return false unless Setting[:dynflow_enable_console]

  if Setting[:dynflow_console_require_auth]
    return false if Time.now.to_i > expires_at.to_i
    user = User.find_by_id(user_id)
    return false if user.nil?
    return true if user.admin?

    # users with unlimited edit_foreman_tasks can operate with the
    # console no matter what task it is…
    edit_permission = Permission.where(:name => :edit_foreman_tasks, :resource_type => ForemanTasks::Task.name).first
    if user.filters.joins(:filterings).unlimited.where('filterings.permission_id' => edit_permission).first
      return true
    end

    # limited users can operate only on specific tasks they are authorized to
    dynflow_id = rack_request.path_info[/^\/([\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12})/,1]
    unless dynflow_id.empty?
      task = ForemanTasks::Task::DynflowTask.where(:external_id => dynflow_id).first
    end
    if task
      begin
        User.current = user
        return Authorizer.new(user).can?(:edit_foreman_tasks, task)
      ensure
        User.current = nil
      end
    else
      return false
    end
  else
    return true
  end
end
```